### PR TITLE
Export packing outputs as a tiff file

### PR DESCRIPTION
**Pull request by mogres**
Problem
=======
See #165

Solution
========
1. The multi-channel image is represented as a numpy array with dimensions `C x X x Y x Z`. Each channel represents a single type of packed ingredient or compartment.
1. For `single_sphere` ingredients, the `ImageWriter` class creates a mask based on the position and radius of the sphere, and sets all pixels within this mask to 255. Other ingredient types are currently unsupported.
2. For compartments represented by meshes, the `ImageWriter` class creates a hollow or filled mask based on the provided options, and sets all pixels within this mask to 255.
3. We can also convolve the image with a point-spread function as needed. Currently available options are "gaussian" and "box".
4. We use the `aicsimagio` package to export the packing image as an `ome.tiff` file. The camera is set looking down the `-z` axis similar to a microscopy assay.

with @meganrm 

## Type of change
* New feature (non-breaking change which adds functionality)

Change summary:
---------------
* added the `ImageWriter` class to export packing outputs as tiff images
* added options in the packing config to edit the output image properties, including:
    1.  the voxel size (in units of grid spacing) OR image size (voxels in each dimension)
    2. hollow or filled mesh representation

Steps to Verify:
----------------
1. `conda activate autopack`
5. `pack -r cellpack/tests/recipes/v2/test_image_export.json -c cellpack/tests/packing-configs/test_image_config.json`

Example packing of peroxisomes in the average nuclear and cell shape:
`pack -r examples/recipes/v2/peroxisomes.json -c cellpack/tests/packing-configs/test_image_config.json`

Screenshots:
-----------------------
image
image
image

Keyfiles:
-----------------------
1. `cellpack/autopack/writers/ImageWriter.py`

### CODE REVIEW

**Comment by mogres** [comment]
- [ ] Include a setting to create hollow masks for spheres.
- [ ] Check with matheus about creating peroxisome meshes

**Comment by meganrm** [comment]
It's hard to be sure, but looking at the screenshots of the magenta spheres, it seems like the image is flipped

**Comment by mogres** [comment]
> It's hard to be sure, but looking at the screenshots of the magenta spheres, it seems like the image is flipped

Yeah, I changed the dimension order in a later commit. Let me update the screenshots

**Review by aditnath — APPROVED** [review]

**Review by rugeli — APPROVED** [review]
I installed aicsimageio and am able to see the figures and the tiff image. Not so sure about what circular mask is, but the changes look good to me!

File: cellpack/autopack/writers/ImageWriter.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+from pathlib import Path
+import numpy
+from scipy.ndimage import convolve
+
+"""
+ImageWriter provides a class to export cellpack packings as tiff images
+"""
+
+
+class ImageWriter:
```
Comment (toloudis) [inline cellpack/autopack/writers/ImageWriter.py:11]: I would suggest that this class should be called Voxelizer or Voxelization etc.  Writing the image out is only a tiny part of what is in here!

File: cellpack/autopack/writers/ImageWriter.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            image = numpy.transpose(image, axes=(3, 0, 2, 1))
+        return image
+
+    def create_voxelization(self):
+        """
+        Creates a voxelized representation of the current scene
+        """
+        channel_colors = []
+
+        for pos, rot, ingr, _ in self.env.molecules:
```
Comment (toloudis) [inline cellpack/autopack/writers/ImageWriter.py:180]: these two for loops (over molecules and compartments) might be more properly class members of whatever type self.env is.  Then you could call self.env.collect_voxelizations(self.image_data) and the function would fill in the image_data object.

**Pull request merged**